### PR TITLE
Additional ETL Improvements

### PR DESCRIPTION
**Pull request by pomadchin**
- refactored configs, to allow etl config parsing functions usage in client projects
- refactored EtlConf case class (removed unused vals)

### CODE REVIEW

**Comment by lossyrob** [comment]
What's an example of what this enables? do we need documentation additions for this?

**Comment by pomadchin** [comment]
@lossyrob it allows at least to re-use json reading function, nothing too critical, but probably it makes sense to use it in integration tests project; 

The important change is `class EtlConf(val input: Input, val output: Output, val inputProfile: Option[BackendProfile] = None, val outputProfile: Option[BackendProfile] = None)` => `class EtlConf(val input: Input, val output: Output)`

File: spark-etl/src/main/scala/geotrellis/spark/etl/config/DefaultEtlConf.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package geotrellis.spark.etl.config
+
+import geotrellis.spark.etl.config.json._
+
+import org.apache.spark.SparkContext
+import com.github.fge.jackson.JsonLoader
+import spray.json._
+
+trait DefaultEtlConf extends ConfigParse {
```
Comment (lossyrob) [inline spark-etl/src/main/scala/geotrellis/spark/etl/config/DefaultEtlConf.scala]: The name default is confusing. Should it be BaseEtlConfig? Perhaps ask the room, if I'm the only one who thinks so then perhaps we keep it as is

Comment (pomadchin) [inline spark-etl/src/main/scala/geotrellis/spark/etl/config/DefaultEtlConf.scala]: @lossyrob I think that the word base is better, though for some reason I chose Default (guess it's spray json influence )

**Pull request merged**